dds/DCPS/FilterEvaluator.cpp

```diff
@@ -125,23 +125,21 @@ FilterEvaluator::SerializedForEval::lookup(const char* field) const
     return iter->second;
   }
   Message_Block_Ptr mb (serialized_->duplicate());
-  Serializer ser(mb.get(),
-    cdr_ ? Encoding::KIND_XCDR1 : Encoding::KIND_UNALIGNED_CDR, swap_);
-  if (cdr_) {
-    EncapsulationHeader encap;
-    if (!(ser >> encap)) {
+  Serializer ser(mb.get(), Encoding::KIND_UNALIGNED_CDR, swap_);
+  EncapsulationHeader encap;
+  if (!(ser >> encap)) {
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR ")
+      ACE_TEXT("%FilterEvaluator::SerializedForEval::lookup: ")
+      ACE_TEXT("deserialization of encapsulation header failed.\n")));
```

Comment:
The `EncapsulationHeader` deserialization and `to_encoding` error cases really should return, but I don't know what that would look like it it wasn't built for that in the first place.